### PR TITLE
Docs/docs refactor initial pass

### PR DESCRIPTION
**Pull request by girayimprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
An initial pass over some docs. Fixed up some language. Revised some suggested use. Removed most of the content from the landing page, it will be revamped.

I left the pre-alpha disclaimers for now, these will be updated before release.
#### Tests
How did you test these changes prior to submitting this pull request?
What automated tests are included in this PR?
#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: SpatialGDK/Documentation/features.md [hunk 1]
```diff
@@ -48,6 +46,10 @@ Actor property handover is a new feature we’re introducing with the SpatialOS
 
 `Singleton Actors` is a new term that describes Actors which contain global state and logic.  Only one of these Actors exists in a game world. The SpatialOS Unreal GDK supports both client accessible and server only singletons (respectively known as public and private singletons). For more information see the documentation on [Singleton Actors]({{urlRoot}}/content/singleton-actors).
 
+## Cross-server RPCs
+
+`Cross-server RPCs` is a new type of RPC that functions between two servers. These will enable a server without authority over an Actor to communicate with the server which has authority and is able to manipulate the Actor. For more information, see [Cross-server RPC]({{urlRoot}}/content/dummy-link) documentation.
```
Comment (girayimprobable) [inline SpatialGDK/Documentation/features.md]: The link will be updated once doc is ready. @danielimprobable will work on the doc.

File: README.md [hunk 2]
```diff
@@ -1,34 +1,28 @@
-# The SpatialOS GDK for Unreal (pre-alpha)
+# The SpatialOS GDK for Unreal Engine (Alpha)
```
Comment (ElleEss) [inline README.md]: I've had a little bit of a rejig to make the readme less long and wordy - also a few questions for you @girayimprobable.
Rather than try to add this as comments, here (which I think would be tricky)  I have added your amendments to mine in the GDoc here: https://docs.google.com/document/d/1tu_VEPJtfTl6F0r8QswuVvKtoDQyXovr9AOV4Uj3fkc/edit
I hope that this is OK.

File: SpatialGDK/Documentation/index.md [hunk 3]
```diff
@@ -6,79 +6,6 @@ Welcome to the SpatialOS and Unreal GDK community.
 
 Our SpatialOS community shapes the functionality of the Unreal GDK, so make sure you are part of the conversation. Come and find us on [Discord](https://discordapp.com/channels/311273633307951114/339471548647866368) or talk to us on the **Feedback** section in our [forums](https://forums.improbable.io/) (use the **unreal-gdk** tag - [this link](https://forums.improbable.io/new-topic?category=Feedback&tags=unreal-gdk) takes you there and pre-fills the category and tag). You can also [sign up for SpatialOS Unreal GDK updates](http://go.pardot.com/l/169082/2018-06-15/27ld2t) direct to your mailbox.
```
Comment (ElleEss) [inline SpatialGDK/Documentation/index.md]: Ernie is working on the welcome page / index page for this - I will incorporate this first pass into the GDoc.
I think, like Unity, we can use the same text for the welcome/index page for the initial release and then review this? Does this work OK?
I have added your edits to the Google doc here, so Ernie knows what we are working from: https://docs.google.com/document/d/16pmz7wNgZzE2VPhDwGxc5pyXtgt-h-DPlEqCGsRKvXI/edit#

File: SpatialGDK/Documentation/known-issues.md [hunk 4]
```diff
@@ -9,13 +9,9 @@ Known issue = any major user-facing bug or lack of user-facing feature that:
...
 | Dynamic component replication. | 2018-07-03 | [UNR-366](https://improbableio.atlassian.net/browse/UNR-366) | Mimic behavior with static components. |
 | Components with replicated sub-objects fail. | 2018-07-03 |[UNR-417](https://improbableio.atlassian.net/browse/UNR-417) | Make all components top level. |
 | You can't have multiple replicated components of the same type on the same Actor. | 2018-07-03 | [UNR-416](https://improbableio.atlassian.net/browse/UNR-416) | None |
-| User code must be IWYU-compliant, meaning you must import all the headers you use.| 2018-07-13 |[UNR-380](https://improbableio.atlassian.net/browse/UNR-380) | None |
 | Schema generator fails when the destination file is locked. The workflow is less than ideal as you can't run it again until you restart the Unreal Editor. | 2018-07-16 | [UNR-350](https://improbableio.atlassian.net/browse/UNR-350) | Ensure destination files/folders are unlocked. |
 | Deleting entities directly from the Inspector causes undefined behavior. | 2018-07-17 | [UNR-425](https://improbableio.atlassian.net/projects/UNR/issues/UNR-425) | None |
 | We don't support listen servers. | 2018-07-30 | | Use dedicated servers instead. |
```
Comment (ElleEss) [inline SpatialGDK/Documentation/known-issues.md]: Cool! Have updated the GDoc to match this.

(FYI): https://docs.google.com/document/d/127kle0sDC0wJkd_wrOaDidzykxhxsQg0vTvyqPst0Ck/edit

File: README.md [hunk 5]
```diff
@@ -1,34 +1,28 @@
...
 
 ## Recommended use
-We are releasing the GDK in [pre-alpha](https://docs.improbable.io/reference/latest/shared/release-policy#maturity-stages) so we can react to feedback and iterate on development quickly. To facilitate this, during our pre-alpha stage we don't have a formal deprecation cycle for APIs and workflows. This means that everything and anything can change. In addition, documentation is limited and some aspects of the GDK are not optimized.
+We are releasing the GDK in [alpha](https://docs.improbable.io/reference/latest/shared/release-policy#maturity-stages) so we can react to feedback and iterate on development quickly. To facilitate this, during our alpha stage we don't have a formal deprecation cycle for APIs and workflows. This means that everything and anything can change. In addition, documentation is limited and some aspects of the GDK are not optimized.
 
-Given this, we don’t recommend using the the GDK for Unreal for games in the production phase just yet.
+Given this, for now we recommend using the GDK in projects in early production or prototype stage. This would ensure that your project's requirements are in line with the GDK's timeline.
```
Comment (anne-edwards) [inline README.md]: Add "the", so "projects in the early production or prototype stage"

Also "this would ensure" -> "this ensures"

File: SpatialGDK/Documentation/features.md [hunk 6]
```diff
@@ -2,9 +2,9 @@
 
 # Feature list
```
Comment (ElleEss) [inline SpatialGDK/Documentation/features.md]: LAURA - NOTE TO SELF - not looked at this yet!

File: SpatialGDK/Documentation/features.md [hunk 7]
```diff
@@ -32,13 +32,13 @@ The SpatialOS Unreal GDK also supports:
...
-## Static subobject replication
+## Subobject replication
 
-The SpatialOS Unreal GDK currently supports only one instance of each static subobject type when replicating objects owned by an Actor. This includes components. For example, multiple Actors can have a `FooComponent` but no Actor can have two `FooComponent`s.
+The GDK currently supports only one instance of each static subobject type when replicating objects owned by an Actor. This includes components. For example, multiple Actors can have a `FooComponent` but no Actor can have two `FooComponent`s.
 
 Properties and RPCs within subobjects have the same support as an Actor’s properties and RPCs.
 
-> The SpatialOS Unreal GDK does not yet support dynamic subobject replication.
+The SpatialOS Unreal GDK does not yet support dynamic subobject replication.
```
Comment (anne-edwards) [inline SpatialGDK/Documentation/features.md]: SpatialOS Unreal GDK -> SpatialOS GDK for Unreal Engine

File: SpatialGDK/Documentation/index.md [hunk 8]
```diff
@@ -6,79 +6,6 @@ Welcome to the SpatialOS and Unreal GDK community.
...
 
-#### What's the Unreal GDK all about?
+#### What is SpatialOS GDK for Unreal Engine?
```
Comment (anne-edwards) [inline SpatialGDK/Documentation/index.md]: Add "the"

File: SpatialGDK/Documentation/features.md [hunk 9]
```diff
@@ -48,6 +48,10 @@ Actor property handover is a new feature we’re introducing with the SpatialOS
 
 `Singleton Actors` is a new term that describes Actors which contain global state and logic.  Only one of these Actors exists in a game world. The SpatialOS Unreal GDK supports both client accessible and server only singletons (respectively known as public and private singletons). For more information see the documentation on [Singleton Actors]({{urlRoot}}/content/singleton-actors).
 
+## Cross-server RPCs
+
+`Cross-server RPCs` is a new type of RPC that functions between two servers. These will enable a server without authority over an Actor to communicate with the server which has authority and is able to manipulate the Actor. For more information, see [Cross-server RPC]({{urlRoot}}/content/dummy-link) documentation.
```
Comment (anne-edwards) [inline SpatialGDK/Documentation/features.md]: Change the first sentence into the plural, i.e. "`Cross-server RPCs` are a new type of RPC that function between two servers."

These will enable -> These enable

Also add "the" before [Cross-server RPC]({{urlRoot}}/content/dummy-link)

File: SpatialGDK/Documentation/features.md [hunk 10]
```diff
@@ -48,6 +48,10 @@ Actor property handover is a new feature we’re introducing with the SpatialOS
 
 `Singleton Actors` is a new term that describes Actors which contain global state and logic.  Only one of these Actors exists in a game world. The SpatialOS Unreal GDK supports both client accessible and server only singletons (respectively known as public and private singletons). For more information see the documentation on [Singleton Actors]({{urlRoot}}/content/singleton-actors).
```
Comment (danielimprobable) [inline SpatialGDK/Documentation/features.md]: SpatialOS Unreal GDK -> SpatialOS GDK for Unreal Engine

File: README.md [hunk 11]
```diff
@@ -1,34 +1,28 @@
...
 
-#### What's the GDK for Unreal all about?
-The SpatialOS GDK for Unreal is an Unreal Engine 4 (UE4) [plugin](https://docs.unrealengine.com/en-us/Programming/Plugins) made by Improbable. You can use the GDK for Unreal to integrate persistent multiplayer worlds into your Unreal game using the SpatialOS platform. Find out more about the GDK for Unreal in [our blogpost](https://improbable.io/games/blog/spatialos-unreal-gdk-pre-alpha).
+#### What is SpatialOS GDK for Unreal Engine?
+The SpatialOS GDK for Unreal Engine is an Unreal Engine 4 (UE4) [plugin](https://docs.unrealengine.com/en-us/Programming/Plugins) made by Improbable. You can use the GDK to make use of multiple Unreal servers in a single game instance using the SpatialOS platform, unlocking the ability to implement large scale, complex and persistent worlds. Find out more about the GDK in [our blogpost](https://improbable.io/games/blog/spatialos-unreal-gdk-pre-alpha).
```
[the remark this answers is not in the document]
Comment (ElleEss) [reply README.md]: Added to GDoc for second pass - Giray's gonna pop this into the repo.

File: README.md [hunk 12]
```diff
@@ -1,34 +1,28 @@
...
+Welcome to the SpatialOS and GDK (Game Development Kit) for Unreal community.
 
 Our SpatialOS community shapes the functionality of the GDK for Unreal, so make sure you are part of the conversation. Come and find us on [Discord](https://discordapp.com/channels/311273633307951114/339471548647866368) or talk to us on the **Feedback** section in our [forums](https://forums.improbable.io/) (use the **unreal-gdk** tag - [this link](https://forums.improbable.io/new-topic?category=Feedback&tags=unreal-gdk) takes you there and pre-fills the category and tag). You can also [sign up for SpatialOS GDK for Unreal updates](http://go.pardot.com/l/169082/2018-06-15/27ld2t) direct to your mailbox.
 
-#### What's the GDK for Unreal all about?
-The SpatialOS GDK for Unreal is an Unreal Engine 4 (UE4) [plugin](https://docs.unrealengine.com/en-us/Programming/Plugins) made by Improbable. You can use the GDK for Unreal to integrate persistent multiplayer worlds into your Unreal game using the SpatialOS platform. Find out more about the GDK for Unreal in [our blogpost](https://improbable.io/games/blog/spatialos-unreal-gdk-pre-alpha).
+#### What is SpatialOS GDK for Unreal Engine?
```
[the remark this answers is not in the document]
Comment (ElleEss) [reply README.md]: Added to GDoc - Giray's gonna pop this into the repo as-is.

File: SpatialGDK/Documentation/known-issues.md [hunk 13]
```diff
@@ -9,13 +9,9 @@ Known issue = any major user-facing bug or lack of user-facing feature that:
...
-| Adaptive Unity builds are not supported. | 2018-07-28 | | Disable `bUseAdaptiveUnityBuild` following [Epic’s documentation](https://docs.unrealengine.com/en-US/Programming/UnrealBuildSystem/Configuration). |
 | Dynamic component replication. | 2018-07-03 | [UNR-366](https://improbableio.atlassian.net/browse/UNR-366) | Mimic behavior with static components. |
 | Components with replicated sub-objects fail. | 2018-07-03 |[UNR-417](https://improbableio.atlassian.net/browse/UNR-417) | Make all components top level. |
 | You can't have multiple replicated components of the same type on the same Actor. | 2018-07-03 | [UNR-416](https://improbableio.atlassian.net/browse/UNR-416) | None |
-| User code must be IWYU-compliant, meaning you must import all the headers you use.| 2018-07-13 |[UNR-380](https://improbableio.atlassian.net/browse/UNR-380) | None |
 | Schema generator fails when the destination file is locked. The workflow is less than ideal as you can't run it again until you restart the Unreal Editor. | 2018-07-16 | [UNR-350](https://improbableio.atlassian.net/browse/UNR-350) | Ensure destination files/folders are unlocked. |
```
Comment (danielimprobable) [inline SpatialGDK/Documentation/known-issues.md]: This is no longer true, I've validated on my machine and this should be closed.

File: README.md [hunk 14]
```diff
@@ -1,34 +1,28 @@
...
+The SpatialOS GDK for Unreal Engine is an Unreal Engine 4 (UE4) [plugin](https://docs.unrealengine.com/en-us/Programming/Plugins) made by Improbable. You can use the GDK to make use of multiple Unreal servers in a single game instance using the SpatialOS platform, unlocking the ability to implement large scale, complex and persistent worlds. Find out more about the GDK in [our blogpost](https://improbable.io/games/blog/spatialos-unreal-gdk-pre-alpha).
 
 
 * License: use of the contents of this repository is subject to the [Improbable license](https://docs.improbable.io/unreal/latest/license) </br>
 (Your access to and use of the Unreal Engine is governed by the [Unreal Engine End User License Agreement](https://www.unrealengine.com/en-US/previous-versions/udk-licensing-resources?sessionInvalidated=true). Please ensure that you have agreed to those terms before you access or use the Unreal Engine.)
-* Version: pre-alpha (for evaluation only)
-* Unreal Game Development Kit repository: [github.com/spatialos/UnrealGDK](https://github.com/spatialos/UnrealGDK)
+* Version: Alpha (Stability and performance improvements pending)
```
[the remark this answers is not in the document]
Comment (ElleEss) [reply README.md]: added to GDoc

File: SpatialGDK/Documentation/known-issues.md (hunk 4 above)
Comment (joshuahuburn) [reply SpatialGDK/Documentation/known-issues.md]: Based on the ToC discussion, this GitHub PR will be the source of truth so updates here will be reflected back later on.

Comment (joshuahuburn) [inline SpatialGDK/Documentation/known-issues.md]:
```suggestion
| We don't support listen servers. | 2018-07-30 | | Use dedicated servers instead. |
| Server travel does not work in PIE. | 2018-07-30 | | Server travel work must be done with external or managed workers |
```
Long story short, since we only support `dedicated server` we can't use server travel in PIE since this requires multiple processes and this is not supported for dedicated servers...
Additionally seamless server travel is not supported in PIE in default Unreal anyway.

File: SpatialGDK/Documentation/features.md [hunk 15]
```diff
@@ -32,13 +32,13 @@ The SpatialOS Unreal GDK also supports:
 * Replicated Blueprint events
 * RPC validation
 
-## Static subobject replication
+## Subobject replication
 
-The SpatialOS Unreal GDK currently supports only one instance of each static subobject type when replicating objects owned by an Actor. This includes components. For example, multiple Actors can have a `FooComponent` but no Actor can have two `FooComponent`s.
+The GDK currently supports only one instance of each static subobject type when replicating objects owned by an Actor. This includes components. For example, multiple Actors can have a `FooComponent` but no Actor can have two `FooComponent`s.
```
Comment (ElleEss) [inline SpatialGDK/Documentation/features.md]: This includes components.
->
This includes and Actor's components.

(Note from editor - in the para above we are talking about SpatialOS components, so we should be absolutely clear - I know you didn't edit this, Giray but I just spotted the potential lack of clarity.)

File: SpatialGDK/Documentation/features.md (hunk 7 above)
Comment (ElleEss) [reply SpatialGDK/Documentation/features.md]: Good spot!
"SpatialOS Unreal GDK"
->
"GDK" 

[Suggest going for the shorter name as we are in the GDK docs - it's easier to read - as Giray said about a year ago! (He was right!)]

File: SpatialGDK/Documentation/features.md (hunk 10 above)
Comment (ElleEss) [reply SpatialGDK/Documentation/features.md]: Suggest "The GDK" as we know we are in the Unreal GDK.  (I think we have been all round the houses on this naming and come back to where we started! :-) )

File: README.md (hunk 5 above)
Comment (ElleEss) [reply README.md]: Added in the GDocs

File: SpatialGDK/Documentation/features.md (hunk 1 above)
Comment (ElleEss) [reply SpatialGDK/Documentation/features.md]: The linting is fierce! It won't let you put this non-link in. Suggest adding this as a "TODO" comment and I will add a note to self to add this.
So:
"[//]: # (TODO: Add link to the Cross-server RPC doc)"
I have added a TODO note in the Doc of Docs Toc master list.

File: SpatialGDK/Documentation/features.md [hunk 16]
```diff
@@ -48,6 +48,10 @@ Actor property handover is a new feature we’re introducing with the SpatialOS
...
 
+## Cross-server RPCs
+
+`Cross-server RPCs` is a new type of RPC that functions between two servers. These will enable a server without authority over an Actor to communicate with the server which has authority and is able to manipulate the Actor. For more information, see [Cross-server RPC]({{urlRoot}}/content/dummy-link) documentation.
+
 ## Networking switch
 
-We’ve implemented a way to switch between the native Unreal networking and SpatialOS networking via an editor tickbox. This can be found inside `Play` dropdown option in the toolbar under `Spatial Networking`. This is useful to test valid functionality on a single server, and performance. See the documentation on the networking switch on the [Troubleshooting]({{urlRoot}}/content/troubleshooting) page.
+We’ve implemented a convenient way to switch between the native Unreal networking and SpatialOS networking via an editor tickbox. This can be found inside `Play` dropdown option in the toolbar under `Spatial Networking`. This is useful to test valid functionality and performance on a single server. See the documentation on the networking switch on the [Troubleshooting]({{urlRoot}}/content/troubleshooting) page.
```
Comment (ElleEss) [inline SpatialGDK/Documentation/features.md]: We’ve implemented a convenient way to switch between the native Unreal networking and SpatialOS networking via an editor tickbox. This can be found inside `Play` dropdown option in the toolbar under `Spatial Networking`. This is useful to test valid functionality and performance on a single server. See the documentation on the networking switch on the [Troubleshooting]({{urlRoot}}/content/troubleshooting) page.

->(This needs tech writer-ing :-) )

The GDK offers you a convenient way to switch between native Unreal networking and SpatialOS networking via a checkbox in the Unreal Editor. This is useful to test valid functionality and performance on a single server. See the documentation on the networking switch on the [Troubleshooting]({{urlRoot}}/content/troubleshooting) page.
To switch networking to SpatialOS, from the Editor toolbar menu, select **Spatial Networking** > **Play** > **????**
[Editor note: is the "????" an option from "Play"? or a pair of options??)

File: SpatialGDK/Documentation/index.md (hunk 8 above)
Comment (ElleEss) [reply SpatialGDK/Documentation/index.md]: Added in the GDoc.

File: SpatialGDK/Documentation/known-issues.md (hunk 4 above)
Comment (ElleEss) [reply SpatialGDK/Documentation/known-issues.md]: Yep - ignore this - removing the link.

**Comment by ElleEss** [comment]
The features.md edits do need adding!

File: SpatialGDK/Documentation/known-issues.md [hunk 17]
```diff
@@ -9,13 +9,10 @@ Known issue = any major user-facing bug or lack of user-facing feature that:
...
 | Deleting entities directly from the Inspector causes undefined behavior. | 2018-07-17 | [UNR-425](https://improbableio.atlassian.net/projects/UNR/issues/UNR-425) | None |
 | We don't support listen servers. | 2018-07-30 | | Use dedicated servers instead. |
-| Stably-named replicated actors cannot be referred to by their path | 2018-08-10 | [UNR-473](https://improbableio.atlassian.net/projects/UNR/issues/UNR-473) | None |
```
Comment (m-samiec) [inline SpatialGDK/Documentation/known-issues.md]: Why was this removed? This is still valid.

Comment (girayimprobable) [reply SpatialGDK/Documentation/known-issues.md]: My bad

File: SpatialGDK/Documentation/features.md (hunk 16 above)
Comment (girayimprobable) [reply SpatialGDK/Documentation/features.md]: I've reworded your version - we want people to use this by default and only uncheck if something is wrong. Feel free to rephrase my version after I commit.
